loginov-rocks/nodejs-aws-be
# Task9

**Note! The app deployed on Sunday Dec 13 only till Dec 17 to reduce costs.**

- Task: https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task9-bff/task.md
- API:
  - BFF Product: http://loginov-rocks-bff-api-dev.eu-central-1.elasticbeanstalk.com/product
  - Original Product: https://d7667crufa.execute-api.eu-central-1.amazonaws.com/dev/products
  - BFF Cart: http://loginov-rocks-bff-api-dev.eu-central-1.elasticbeanstalk.com/cart/profile/cart
  - Original Cart: http://loginov-rocks-cart-api-dev.eu-central-1.elasticbeanstalk.com/api/profile/cart
- Product Swagger (use to test cache with the help of product creation): https://editor.swagger.io/?url=https://raw.githubusercontent.com/loginov-rocks/nodejs-aws-be/main/product-service/openapi.yaml
- How to test cache:
  - Open http://loginov-rocks-bff-api-dev.eu-central-1.elasticbeanstalk.com/product
  - Go to Product Swagger and create product with POST /product
  - Open http://loginov-rocks-bff-api-dev.eu-central-1.elasticbeanstalk.com/product again (in less then 2 mins) - there should be no new product
  - Wait 2 mins, open products list once again - there should be a new product
- Check the implementation:
  - https://github.com/loginov-rocks/nodejs-aws-be/blob/task9/bff-service/src/app.controller.ts
  - https://github.com/loginov-rocks/nodejs-aws-be/blob/task9/bff-service/src/discovery.service.ts
  - https://github.com/loginov-rocks/nodejs-aws-be/blob/task9/bff-service/src/proxy.service.ts
- FE: http://drh6q7egmt2il.cloudfront.net
- Evaluation criteria:
- Additional (optional) tasks
How to test:
    - Get products list
    - Create new product
    - Get products list - result shouldn’t have new product
    - Wait more than 2 minutes
    - Get products list - result should have new product

Updated architecture
![Architecture](https://raw.githubusercontent.com/loginov-rocks/nodejs-aws-be/task9/docs/Architecture.png)
